llpc/lower/llpcSpirvLowerInternalLibraryIntrinsicUtil.h

```diff
@@ -0,0 +1,62 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2023 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcSpirvLowerInternalLibraryIntrinsicUtil.cpp
+ * @brief LLPC header file: utilities for lowering common internal library intrinsics.
+ ***********************************************************************************************************************
+ */
+
+#include "llvm/ADT/DenseMap.h"
+
+namespace lgc {
+class Builder;
+}
+
+namespace llvm {
+class Function;
+class StringRef;
+} // namespace llvm
+
+namespace Llpc {
+class InternalLibraryIntrinsicUtil {
+public:
+  typedef void (*LibraryFuncPtr)(llvm::Function *, lgc::Builder *);
+
+  struct LibraryFunctionTable {
+    llvm::DenseMap<llvm::StringRef, LibraryFuncPtr> m_libFuncPtrs;
+    LibraryFunctionTable();
+    static const LibraryFunctionTable &get() {
+      static LibraryFunctionTable instance;
+      return instance;
+    }
+  };
+
+private:
+  static void createLaneIndex(llvm::Function *func, lgc::Builder *builder);
+  static void createLaneCount(llvm::Function *func, lgc::Builder *builder);
+  static void createHalt(llvm::Function *func, lgc::Builder *builder);
```

Comment:
Since these are `static` anyway, perhaps we can just make them static global functions in the .cpp file and avoid having them in the header, to make the header file smaller?